pkg/framework/test/etcd.go

```diff
@@ -1,42 +1,54 @@
 package test
 
 import (
+	"io/ioutil"
+	"os"
 	"os/exec"
 
-	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
 	"github.com/onsi/gomega/gbytes"
 	"github.com/onsi/gomega/gexec"
 )
 
 // Etcd knows how to run an etcd server. Set it up with the path to a precompiled binary.
 type Etcd struct {
 	// The path to the etcd binary
-	Path    string
-	session *gexec.Session
+	Path           string
+	EtcdURL        string
+	session        *gexec.Session
+	stdOut         *gbytes.Buffer
+	stdErr         *gbytes.Buffer
+	tempDirManager TempDirManager
 }
 
 // Start starts the etcd, and returns a gexec.Session. To stop it again, call Terminate and Wait on that session.
-func (e *Etcd) Start(etcdURL string, datadir string) error {
+func (e *Etcd) Start() error {
+	e.tempDirManager = &tempDirManager{}
+
+	dataDir := e.tempDirManager.Create()
+
 	args := []string{
+		"--debug",
```

Comment:
I have the feeling that this is going to be painful to maintain?

Reply:
If this is only about using `--debug` or not, we can totally remove that line. If it's about management of arguments (potentially for different versions, ...), we believe we have already covered that in [the other comment](https://github.com/kubernetes/kubectl/pull/149#discussion_r153734672).